app/src/main/java/com/example/ithaca_transit_android_v2/states/SearchState.kt

```diff
@@ -0,0 +1,78 @@
+package com.example.ithaca_transit_android_v2.states
+
+import com.example.ithaca_transit_android_v2.models.Coordinate
+import com.example.ithaca_transit_android_v2.models.Location
+
+//Parent Search State Class - Contains Default parameters
+sealed class SearchState(
+    val startLocation: Location = Location("", Coordinate(0.0,0.0),""),
+    val endLocation: Location = Location("", Coordinate(0.0,0.0),"") ,
+    val favDestinations: List<Location> = mutableListOf(),
+    val recentDestinations: List<Location> = mutableListOf()
+){}
+
+//Default UI when launching app. Just displays a search bar (no interactions so far)
+class LaunchState(
+): SearchState(){}
+
+//First time using the app, after clicking on the search bar, there will be nothing
+//but "Search For a Destination" text and a big magnifying glass
+class EmptyInitSearchState(
+): SearchState(){}
+
+//Click on search bar -> Changes UI by dropping a list of Favorite Destinations and
+//recent destinations
+class InitSearchState(
+    favDestinations : List<Location>,
+    recentDestinations : List<Location>
```

Comment:
typically it's convention to have trailing commas, not sure if it's a syntax error in `kotlin`? @ningning621

Reply:
i don't think trailing commas are possible in kotlin? https://discuss.kotlinlang.org/t/language-feature-request-trailing-commas/2511